### PR TITLE
Make everything pip installable

### PR DESCRIPTION
**Pull request by abhi-mosaic**
This PR touches almost every file, but it is doing very little real editing, and is mainly shuffling files around:
* move all code under `examples/examples`
* change `setup.py` to install everything
* update READMEs and import statements

### CODE REVIEW

File: examples/llm/throughput/run_all_configs.py [hunk 1]
```diff
@@ -257,13 +257,13 @@ def run_config(config, args, project, image, RUN):
     # Define our command
     if streaming_data:
         command = """
-        composer examples/llm/main.py /mnt/config/parameters.yaml
+        composer examples/common/llm/main.py /mnt/config/parameters.yaml
```
Comment (dblalock) [inline examples/llm/throughput/run_all_configs.py:260]: wrong dir?

```suggestion
        composer examples/examples/llm/main.py /mnt/config/parameters.yaml
```

File: examples/llm/throughput/run_all_configs.py [hunk 2]
```diff
@@ -257,13 +257,13 @@ def run_config(config, args, project, image, RUN):
...
         """
     else:
         command = """
-        python examples/llm/convert_c4.py --out_root ./my-copy-c4 --splits val
+        python examples/common/convert_c4.py --out_root ./my-copy-c4 --splits val
 
-        composer examples/llm/main.py /mnt/config/parameters.yaml
+        composer examples/common/llm/main.py /mnt/config/parameters.yaml
```
Comment (dblalock) [inline examples/llm/throughput/run_all_configs.py:266]: also wrong dir?
```suggestion
        composer examples/examples/llm/main.py /mnt/config/parameters.yaml
```

File: examples/llm/train.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+source /workdisk/abhi/examples/venv-new/bin/activate
+composer main.py yamls/mosaic_gpt/125m.yaml train_loader.dataset.split=train_small
```
Comment (dblalock) [inline examples/llm/train.sh:3]: Is this file supposed to be here? It doesn't seem to be referenced anywhere and assumes there's a `venv-new` that something else creates + cleans up

File: examples/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# These requirements are shared for all examples
+# Each example subdir has its own `requirements.txt` with extra deps.
```
Comment (dblalock) [inline examples/requirements.txt:2]: is there a reason the repo-wide requirements aren't in the repo rootdir? I would expect at least the dev requirements (pre-commit, pytest, etc) to not be under examples.

File: setup.py [hunk 5]
```diff
@@ -5,19 +5,17 @@
...
 
 _PACKAGE_NAME = 'mosaicml-examples'
-_PACKAGE_DIR = 'mosaicml_examples'
-_EXAMPLE_SUBDIRS = ('cifar', 'resnet', 'deeplab', 'bert', 'llm')
-
-_project_root = os.path.dirname(os.path.realpath(__file__))
-
-print('_project_root:', _project_root)
+_EXAMPLE_SUBDIRS = ['cifar', 'resnet', 'deeplab', 'bert', 'llm']
+_PACKAGES = find_packages(exclude=['*tests*', '*scripts*'])
```
Comment (dblalock) [inline setup.py:12]: Is the goal still to just have `examples` be the one package? I think this will make any subdir with an __init__.py a package, which might be unintended behavior down the road.

File: examples/resnet/tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
```
Comment (dblalock) [inline examples/resnet/tests/__init__.py:2]: do we want the tests to be a package? If so we should make sure to do this for all test subdirs.

I kinda don't want it to be a package unless there's code sharing across test files so that we have to explicitly opt into importing stuff from test files. Just part of a general defensive policy wrt enforcing modularity.

**Review by dblalock — COMMENTED** [review]
LGTM aside from local nitpicks. The only part that seems weird to me is moving requirements.txt out of the repo root.

File: examples/bert/yamls/main/mcloud_run.yaml [hunk 7]
```diff
@@ -100,26 +102,27 @@ parameters:
...
+#   (Optional) W&B logging
+#   loggers:
+#     wandb:
+#       project:
+#       entity:
+
+#   (Optional) Checkpoint to local filesystem or remote object store
+#   save_interval: 3500ba
+#   save_num_checkpoints_to_keep: 1  # Important, this cleans up checkpoints saved to DISK
+#  save_folder:      # e.g. './{run_name}/ckpt' (local) or 's3://mybucket/mydir/{run_name}/ckpt' (remote)
```
[the remark this answers is not in the document]
Comment (abhi-mosaic) [reply examples/bert/yamls/main/mcloud_run.yaml:122]: Yup, disabling checkpointing by default.

File: examples/llm/train.sh (hunk 3 above)
Comment (abhi-mosaic) [reply examples/llm/train.sh:3]: Yup this is my bad, will remove.

File: examples/requirements.txt (hunk 4 above)
Comment (abhi-mosaic) [reply examples/requirements.txt:2]: Will move the `requirements.txt` to the top level folder

**Review by dblalock — APPROVED** [review]
LGTM except that the Makefile also a line for BERT under the `test` target
